src/qmapshack/gis/search/CSearchLineEdit.cpp

```diff
@@ -0,0 +1,184 @@
+/**********************************************************************************************
+    Copyright (C) 2019 Henri Hornburg hrnbg@t-online.de
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+**********************************************************************************************/
+#include "CSearchLineEdit.h"
+
+CSearchExplanationDialog* CSearchLineEdit::explanationDlg = nullptr;
+
+CSearchLineEdit::CSearchLineEdit(QWidget *parent)
+    : QLineEdit (parent)
+{
+    actionClearFilter = new QAction(QIcon(":/icons/32x32/Filter.png"), tr("Clear Filter"), this);
+    actionHelp = new QAction(QIcon(":/icons/32x32/CSrcUnknown.png"), tr("Open Help Window"), this);
+    actionSetupFilter = new QAction(QIcon(":/icons/32x32/Apply.png"), tr("Setup Filter"), this);
+    actionError = new QAction(QIcon(":/icons/32x32/Attention.png"), tr("Error parsing search"), this);
+    actionAutoProperty = new QAction(QIcon(":/icons/32x32/Hint.png"), tr("Auto selected property"), this);
+
+    actionNameOnly = new QAction(tr("Name Only"), this);
+    actionNameOnly->setCheckable(true);
+    actionCompleteText = new QAction(tr("Complete Text"), this);
+    actionCompleteText->setCheckable(true);
+    actionCaseSensitive = new QAction(tr("Case Sensitive"), this);
+    actionCaseSensitive->setCheckable(true);
+
+    addAction(actionClearFilter, QLineEdit::TrailingPosition);
+    addAction(actionHelp, QLineEdit::TrailingPosition);
+    addAction(actionSetupFilter, QLineEdit::LeadingPosition);
+
+    connect(actionSetupFilter, &QAction::triggered, this, &CSearchLineEdit::slotSetupSearch);
+    connect(actionNameOnly, &QAction::triggered, this, &CSearchLineEdit::slotSearchNameOnly);
+    connect(actionCompleteText, &QAction::triggered, this, &CSearchLineEdit::slotSearchCompleteText);
+    connect(actionCaseSensitive, &QAction::triggered, this, &CSearchLineEdit::slotCaseSensitive);
+    connect(actionHelp, &QAction::triggered, this, &CSearchLineEdit::slotSearchHelp);
+    connect(actionClearFilter, &QAction::triggered, this, &CSearchLineEdit::slotClearFilter);
+    connect(this, &CSearchLineEdit::textChanged, this, &CSearchLineEdit::slotCreateSearch);
+}
+
+CSearchLineEdit::CSearchLineEdit(QWidget *parent, IGisProject* project, CSearch* search) : CSearchLineEdit(parent)
+{
+    connectedProject=project;
```

Comment:
```
CSearchLineEdit::CSearchLineEdit(....
    : CSearchLineEdit(parent)
    , connectedProject(project)
```

Reply:
This is not possible, since an initialiser for a delegated constructor must appear alone